Separate pressing Enter to open map from pressing Enter to interact

InfoTeddy opened the pull request:
This is a lot of copy-pasted code, but a little bit of copy-pasting never hurt anyone...

The keybind to interact with activity zones and teleporters is now separate from the keybind to open the map, or return to the editor from in-editor playtesting, or restart a time trial. The keybind is now E, and the default controller bind is X. No controller button prompts, but the game didn't have controller button prompts anyways, so whatever.

Doing this now because if people's muscle memory are going to be broken by not being able to spam the map keybind anymore, at least we can help a bit by changing the keybind so they can keep spamming it - their muscle memory is going to be broken anyways.

This option has to be enabled by going to the speedrunner menu options and selecting "interact button". It is disabled by default.

All prompt text needs to be string-interpolated every time they are drawn, because it is possible for people to change which interact button they use in the middle of gameplay, via the in-game options.

Closes #736.

## Legal Stuff:

By submitting this pull request, I confirm that...

- [X] My changes may be used in a future commercial release of VVVVVV
- [X] I will be credited in a `CONTRIBUTORS` file and the "GitHub Friends"
  section of the credits for all of said releases, but will NOT be compensated
  for these changes

flibitijibibo commented:
This gets into 2.4 territory, but I can be overruled here if Terry thinks this is worth breaking the freeze.

TerryCavanagh commented:
Oh, uff, nope, sorry, I don't like this. I very deliberately kept the keybindings simple in VVVVVV and adding E (I guess because it's close to WASD?) as a necessary key really confuses things. Sorry.

A pull request I'd be happy to support would be for custom rebinding for each control in the game, where the default behaviour would be the classic "Enter" as interact key, and people could specify if they wanted to use another key for that instead. Or, less dramatically, I'd also be ok with a speedrunner's option to use E instead of Enter in the speedrunner menu. But I don't like the idea of changing the default here.

flibitijibibo commented:
Yeah, let's try and make the action set rewrite the very first thing we do for 2.4, so this can be made available ASAP.

InfoTeddy commented:
Interesting that you want to keep the keybindings simple; I recall one of the console versions separating Enter to open map from Enter to interact. Although, I don't play the console versions, so correct me if I'm wrong about that.

> I'd also be ok with a speedrunner's option to use E instead of Enter in the speedrunner menu.

Alright; I've added the toggle switch, rebased, and force-pushed. @TerryCavanagh can you take a look at this again?

TerryCavanagh commented:
Ok! Had a read through and that looks good to me, thanks!
